types/jss/index.d.ts

```diff
@@ -1,50 +1,127 @@
+	toJSON(): string;
 }
-
-interface Selectors {
-    [selector: string]: Properties;
+export interface StyleSheet<T> {
+	// Gives auto-completion on the rules declared in `createStyleSheet` without
+	// causing errors for rules added dynamically after creation.
+	classes: {
+		[K in keyof T]: string;
+	} & { [key: string]: string };
+	options: any;
+	linked: boolean;
+	attached: boolean;
+	/**
+	 * Attach renderable to the render tree.
+	 */
+	attach(): this;
+	/**
+	 * Remove renderable from render tree.
+	 */
+	detach(): this;
+	/**
+	 * Add a rule to the current stylesheet.
+	 * Will insert a rule also after the stylesheet has been rendered first time.
+	 */
+	addRule(style: Style, options?: Partial<RuleOptions>): Rule;
+	addRule(name: string, style: Style, options?: Partial<RuleOptions>): Rule;
+	/**
+	 * Create and add rules.
+	 * Will render also after Style Sheet was rendered the first time.
+	 */
+	addRules(styles: { [key: string]: Style }, options?: Partial<RuleOptions>): Rule[];
+	/**
+	 * Get a rule by name.
+	 */
+	getRule(name: keyof T | string): Rule;
+	/**
+	 * Delete a rule by name.
+	 * Returns `true`: if rule has been deleted from the DOM.
+	 */
+	deleteRule(name: string): boolean;
+	/**
+	 * Get index of a rule.
+	 */
+	indexOf(rule: Rule): number;
+	/**
+	 * Link renderable CSS rules from sheet with their corresponding models.
+	 */
+	link(): this;
+	/**
+	 * Update the function values with a new data.
+	 */
+	update(data?: {}): this;
+	update(name?: string, data?: {}): this;
+	/**
+	 * Convert rules to a CSS string.
+	 */
+	toString(options?: { indent?: number }): string;
 }
-
-interface JSS {
-    /**
-     * Retrieve all rules added via JSS, organized by selectors
-     */
-    get(): Selectors;
-
-    /**
-     * Retrieve rules added via JSS for a given selector
-     * @param s CSS selector
-     */
-    get(s: string): Properties;
-
-    /**
-     * Retrieve all rules specified for a given selector (not necessarily added via JSS)
-     * @param s CSS selector
-     */
-    getAll(s: string): Properties;
-
-    /**
-     * Remove all rules added via JSS
-     */
-    remove(): void;
-
-    /**
-     * Remove all rules added via JSS for the given selector
-     */
-    remove(s: string): void;
-
-    /**
-     * Add or extend an existing rule
-     * @param s CSS selector
-     * @param p CSS properties
-     */
-    set(s: string, p: Properties): void;
+export type GenerateClassName<T> = (rule: Rule, sheet?: StyleSheet<T>) => string;
+export interface Style {
+	[key: string]: any;
 }
-
-declare var jss: JSS;
+export interface JSSPlugin {
+	[key: string]: () => Partial<{
+		onCreateRule(name: string, style: Style, options: RuleOptions): Rule,
+		onProcessRule(rule: Rule, sheet: StyleSheet<any>): void,
+		onProcessStyle(style: Style, rule: Rule, sheet: StyleSheet<any>): Style,
+		onProcessSheet(sheet: StyleSheet<any>): void,
+		onChangeValue(value: any, prop: string, rule: Rule): any,
```

Comment:
There is one more: `onUpdate(data, rule, sheet)`

Reply:
I don't see it in `jss` or in http://cssinjs.org/plugins?v=v9.3.3